scalding-commons/src/main/scala/com/twitter/scalding/commons/source/PailSource.scala

```diff
@@ -145,7 +161,7 @@ object PailSource {
   }
 }
 
-class PailSource[T] private (rootPath: String, structure: PailStructure[T], subPaths: Array[List[String]] = null)(implicit conv: TupleConverter[T])
+class PailSource[T] private (rootPath: String, spec: PailSpec, subPaths: Array[List[String]] = Array.empty)(implicit conv: TupleConverter[T])
   extends Source with Mappable[T] {
```

Comment:
can you add with `TypedSink[T]`? Is there any reason why you can't write to this?